### PR TITLE
Add platform Switch defines.

### PR DESCRIPTION
**Pull request by kevin-mccullough**
Adding the RENDERDOC_PLATFORM_SWITCH and RDOC_SWITCH defines in the appropriate places.

### CODE REVIEW

File: renderdoc/common/common.h
Comment (baldurk) [inline renderdoc/common/common.h]: As I mentioned in the other PR, these checks should be using `RDOC_SWITCH`. `RENDERDOC_PLATFORM_SWITCH` should only be used in `globalconfig.h` to be able to define `RDOC_SWITCH`, or in `apidefs.h` since it's a public header and can't include `globalconfig.h`. All other uses should use `RDOC_SWITCH`.

File: renderdoc/driver/gl/gl_dispatch_table.h
Comment (baldurk) [inline renderdoc/driver/gl/gl_dispatch_table.h]: One thing you need to be careful of with these is that `defined(RDOC_SWITCH)` is always true, because we always define it either to `OPTION_ON` or `OPTION_OFF`. You need to use `ENABLED()` or `DISABLED()` as with the other checks.

The reason is mostly to avoid the ambiguity with different conventions where if a macro is defined as 0/1 but then checked with `defined()` it will always be counted as true. The `ENABLED()` and `DISABLED()` checks will fail to compile unless the parameter is defined to either `OPTION_ON` or `OPTION_OFF`, and fail to compile if it's undefined completely.

File: renderdoc/driver/gl/gl_common.h
Comment (baldurk) [inline renderdoc/driver/gl/gl_common.h]: This should use `ENABLED(RDOC_SWITCH)`.

File: renderdoc/driver/gl/gl_hooks.cpp
Comment (baldurk) [inline renderdoc/driver/gl/gl_hooks.cpp]: This should use `ENABLED(RDOC_SWITCH)`.

File: renderdoc/driver/gl/official/cgl.h
Comment (baldurk) [inline renderdoc/driver/gl/official/cgl.h]: Why do you need to make this change to the OS X cgl headers? These are official headers and shouldn't be changed.

**Review by baldurk — CHANGES_REQUESTED** [review]
For future reference closing the PR and re-opening a new one isn't a good idea since it then loses any discussion and history. You can push to the same branch and the PR will update, and then it's easier to see that comments have been addressed.

**Comment by kevin-mccullough** [comment]
Got it.  I was concerned with the commit history being filled with fixes, but seeing as I have some fixes still to do this will be good practice.  Reading your notes, Prepairing-Commits, this seems like a good candidate for squashing and rebasing?

File: renderdoc/driver/gl/gl_dispatch_table.h
Comment (kevin-mccullough) [reply renderdoc/driver/gl/gl_dispatch_table.h]: Sorry, I did mean to change this and overlooked it.

File: renderdoc/common/common.h
Comment (kevin-mccullough) [reply renderdoc/common/common.h]: Sorry, I misread your previous comment and thought you meant to use RENDERDOC_PLATFORM_SWITCH  for the common files.  I will fix this.

**Comment by baldurk** [comment]
Yep I tend to leave it up to PR authors to decide how to manage their branch to avoid being overly proscriptive. The only thing I ask is that each commit compiles, and is formatted - so there are no formatting-only fixes.

Otherwise it's up to you if you squash every time, or keep review feedback as separate commits, or push separate commits until the review is approved then rebase/squash (if you want to do this let me know, otherwise I'm liable to merge before you've squashed).

File: renderdoc/driver/gl/official/cgl.h
Comment (kevin-mccullough) [reply renderdoc/driver/gl/official/cgl.h]: I'll double-check with the team (this one is from before my time).  I can revert it.

**Comment by kevin-mccullough** [comment]
OK.  I did a squash.  Let me know what you think.

**Review by baldurk — APPROVED** [review]
Looks good, thanks!

**Pull request merged**
